### PR TITLE
don't use meta dependencies if they already exist

### PR DESCRIPTION
**Pull request by jsecchiero**
Hi,
can you add this? and would be good if you use also use github release for better versioning.
nice role anyway :)
Jacopo

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ patroni_restapi_username: ""
 patroni_restapi_password: ""
 
 patroni_dcs: etcd
+patroni_dcs_exist: false
```
Comment (kostiantyn-nemchenko) [inline defaults/main.yml]: DCS (or distributed configuration store) is a singular noun, so would be correct to use `patroni_dcs_exists`

File: meta/main.yml [hunk 2]
```diff
@@ -22,14 +22,17 @@ galaxy_info:
   - failover
 dependencies:
   - role: brianshumate.consul
-    when: patroni_dcs == "consul"
+    when: patroni_dcs == "consul" and
+          patroni_dcs_exist == false
```
Comment (kostiantyn-nemchenko) [inline meta/main.yml]: let's make this condition more pythonic and [PEP8 compliant](https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html) by using `... and not patroni_dcs_exists`

**Comment by kostiantyn-nemchenko** [comment]
Hi, Jacopo!
Thank you for your input. I've requested few minor changes which should be addressed before approving this PR. @jsecchiero 
Agree that GitHub releases are needed. I'm going to use them after implementing [Kubernetes-native support](http://patroni.readthedocs.io/en/latest/kubernetes.html) and creating a proper testing strategy for this role.

**Comment by jsecchiero** [comment]
Updated! i hope it's better now.
Jacopo

File: meta/main.yml [hunk 3]
```diff
@@ -22,14 +22,17 @@ galaxy_info:
   - failover
 dependencies:
   - role: brianshumate.consul
-    when: patroni_dcs == "consul"
+    when: patroni_dcs == "consul" and
+          not patroni_dcs_exist
```
Comment (kostiantyn-nemchenko) [inline meta/main.yml]: We're almost there. The variable name `patroni_dcs_exists` declared in `defaults/main.yml` must exactly match the variable name used in `meta/main.yml`

**Comment by jsecchiero** [comment]
another update, missing the 's'. sorry :)
Jacopo

**Comment by kostiantyn-nemchenko** [comment]
Merged. Thanks a lot!

**Pull request merged**